### PR TITLE
Add tab components

### PR DESCRIPTION
**Pull request by torjusti**
I messed around with creating a tab component with sliding underline. More work probably needed especially on fonts and colors.

Screenshot from 2019-07-07 18-54-51

### CODE REVIEW

File: src/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if (selectedTab && selectedTab.current) {
+      setSliderOffset(selectedTab.current.offsetLeft);
+      setSliderWidth(selectedTab.current.clientWidth);
+    }
+  }, [activeTab]);
+
+  return (
+    <div {...rest}>
+      <TabContainer>
+        {React.Children.map(children, (child) => {
```
Comment (oleast) [inline src/components/Tabs/Tabs.tsx]: Could we do this transformation in the component body or a function, It is quite a lot happening inside the JSX content

File: src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState, useRef, FunctionComponent } from 'react';
+import styled from 'styled-components';
+import colors from '../../colors';
```
Comment (oleast) [inline src/components/Tabs/Tabs.tsx]: Would prefer imports based in root-relative paths

File: src/stories/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { storiesOf } from '@storybook/react';
+import { Tabs, Tab } from '../components/Tabs';
+
+const TabsExample = () => {
+  const [tab, setTab] = useState(0);
+
+  return (
+    <div>
+      <Tabs activeTab={tab}>
+        <Tab onClick={() => setTab(0)} tab={0}>
```
Comment (oleast) [inline src/stories/Tabs.tsx]: Could we handle this in a way without manually setting the tab index somehow?

File: src/components/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            });
+          }
+
+          return React.cloneElement(child);
+        })}
+      </TabContainer>
+
+      <SliderBackground />
+
+      <Slider
```
[the remark this answers is not in the document]
Comment (torjusti) [reply src/components/Tabs/Tabs.tsx]: Changed it so that Slider takes `offset` and `width` as props

File: src/components/Tabs/Tabs.tsx (hunk 1 above)
Comment (torjusti) [reply src/components/Tabs/Tabs.tsx]: Moved it into the component body

File: src/stories/Tabs.tsx (hunk 3 above)
Comment (torjusti) [reply src/stories/Tabs.tsx]: Maybe, but I don't know what's the best way of doing it.

Perhaps we could provide an API such as

```javascript
<Tabs>
  <TabPane header="Tab 1">
    Lorem ipsum...
  </TabPane>
</Tabs>
```

in addition to the current one?

**Comment by plusk** [comment]
What's the status on this @torjusti?

**Comment by torjusti** [comment]
@plusk , I cleaned it up a bit so that an additional API of this form is supported:

```
    <Tabs>
        <Tab title="Tab 1">
            Contents of Tab 1.
        </Tab>

        <Tab title="Tab 2">
            Contents of Tab 2.
        </Tab>
    </Tabs>
```

Feel free to give reviews focusing both on the implementation and the visual details - I feel like this doesn't look very online-y yet.

**Pull request merged**